pkg/image-graph-generator/manifests.go

```diff
@@ -0,0 +1,172 @@
+package imagegraphgenerator
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	buildv1 "github.com/openshift/api/build/v1"
+	imagev1 "github.com/openshift/api/image/v1"
+
+	"github.com/openshift/ci-tools/pkg/api"
+)
+
+var (
+	coreScheme   = scheme.Scheme
+	codecFactory = serializer.NewCodecFactory(coreScheme)
+)
+
+func init() {
+	utilruntime.Must(imagev1.AddToScheme(coreScheme))
+	utilruntime.Must(buildv1.AddToScheme(coreScheme))
+}
+
+func (o *Operator) loadManifests(path string) error {
+	err := filepath.Walk(path,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+
+			if ext := filepath.Ext(info.Name()); ext != ".yaml" && ext != ".yml" {
+				return nil
+			}
+
+			d, err := os.ReadFile(path)
+			if err != nil {
+				return err
+			}
+
+			// The YAML file could hold multiple objects separated by `---`
+			dataList := strings.Split(string(d), "---")
```

Comment:
Isn't a better way to unmarshal multi-document yaml file?

Reply:
Nope :(